### ISSUE TITLE
Add an rmsle evaluation metric?

### ISSUE THREAD

**Issue by Carl-McBride-Ellis** [dreamquark-ai/tabnet#228]
## Feature request

The implementation of a root [mean squared logarithmic error regression loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_log_error.html) (`rmsle`) evaluation metric as an out-of-the-box option in TabNet, as it is frequently used in kaggle competitions.
I know one can write a little python class, to do this, but having it as an option would be even easier!

**What is the expected behavior?**

return the rmsle

**What is motivation or use case for adding/changing the behavior?**

This particular metric is often used in kaggle competitions.

**How should this be implemented in your opinion?**

A simple addition+modification based on the code section used to calculate/evaluate the `rmse` 

**Are you willing to work on this yourself?**
Sorry, no, I would probably ruin the whole code...

**Comment by Optimox** [comment]
Thanks @Carl-McBride-Ellis , as discussed in Kaggle discussion this is indeed something interesting we can add. If you have more suggestions about classical metrics we could add by default please add them here in comments!

**Comment by Carl-McBride-Ellis** [comment]
Fantastic; when it is available I shall update and subsequently re-run the kaggle notebook ["TabNet: A very simple regression example"](https://www.kaggle.com/carlmcbrideellis/tabnet-a-very-simple-regression-example) to reflect this option. (I am also sure that the score achieved by TabNet will also improve)

Many thanks!

**Comment by Optimox** [comment]
@Carl-McBride-Ellis "I am also sure that the score achieved by TabNet will also improve" : please note that adding rmsle as default evaluation metric will help for ealrly stopping, but won't change anything in the results of the algorithm itself (will simply change the best epoch). The evaluation metric is not the loss function, but you can pass any loss function to tabnet by providing `loss_fn` in fit params - this would probably help.

**Comment by Carl-McBride-Ellis** [comment]
Duly noted! Question: ¿does `loss_fn` have to be from those listed on [torch.nn](https://pytorch.org/docs/stable/nn.html#loss-functions), or is it possible to use a custom loss fn.?

Many thanks and all the best,
carl

**Comment by Optimox** [comment]
no it does not have to be an official loss from pytorch, but it does have to be written in pytorch, that's the only constraint.

**Comment by Optimox** [comment]
@Carl-McBride-Ellis this PR #248 will close this issue.

You'll be able to update your notebook once the new release 3.0 will be out.
